sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaRulesDefinition.java

```diff
@@ -68,16 +69,17 @@ public void define(Context context) {
       .createRepository(CheckList.REPOSITORY_KEY, Java.KEY)
       .setName("SonarAnalyzer");
     List<Class<?>> checks = CheckList.getChecks();
-    new RulesDefinitionAnnotationLoader().load(repository, checks.toArray(new Class[]{}));
+    new RulesDefinitionAnnotationLoader().load(repository, checks.toArray(new Class[] {}));
     JavaSonarWayProfile.Profile profile = JavaSonarWayProfile.readProfile();
+    Set<String> sonarWayRuleKeys = new HashSet<>(profile.ruleKeys);
```

Comment:
`profile.ruleKeys` is an ArrayList (at least with this current implementation of Gson) so the `contains` method is not very efficient.